zebrad/tests/acceptance.rs

```diff
@@ -1052,3 +1055,170 @@ async fn tracing_endpoint() -> Result<()> {
+
+    // Write a configuration that has our created metrics endpoint_addr
+    let mut config = default_test_config()?;
+    config.metrics.endpoint_addr = Some(listen_addr.parse().unwrap());
+    let dir1 = TempDir::new("zebrad_tests")?.with_config(&mut config)?;
+    let regex1 = format!(r"Opened metrics endpoint at {}", listen_addr);
+
+    // From another folder create a configuration with the same endpoint.
+    // `metrics.endpoint_addr` will be the same in the 2 nodes.
+    // But they will have different Zcash listeners (auto port) and states (ephemeral)
+    let dir2 = TempDir::new("zebrad_tests")?.with_config(&mut config)?;
+
+    check_config_conflict(dir1, regex1.as_str(), dir2, PORT_IN_USE_ERROR)?;
+
+    Ok(())
+}
+
+/// Start 2 zebrad nodes using the same tracing listener port, but different
+/// state directories and Zcash listener ports. The first node should get
+/// exclusive use of the port. The second node will panic with the Zcash tracing
+/// conflict hint added in #1535.
+#[test]
+fn zcash_tracing_conflict() -> Result<()> {
+    zebra_test::init();
+
+    // [Note on port conflict](#Note on port conflict)
+    let port = random_known_port();
+    let listen_addr = format!("127.0.0.1:{}", port);
+
+    // Write a configuration that has our created tracing endpoint_addr
+    let mut config = default_test_config()?;
+    config.tracing.endpoint_addr = Some(listen_addr.parse().unwrap());
+    let dir1 = TempDir::new("zebrad_tests")?.with_config(&mut config)?;
+    let regex1 = format!(r"Opened tracing endpoint at {}", listen_addr);
+
+    // From another folder create a configuration with the same endpoint.
+    // `tracing.endpoint_addr` will be the same in the 2 nodes.
+    // But they will have different Zcash listeners (auto port) and states (ephemeral)
+    let dir2 = TempDir::new("zebrad_tests")?.with_config(&mut config)?;
+
+    check_config_conflict(dir1, regex1.as_str(), dir2, PORT_IN_USE_ERROR)?;
+
+    Ok(())
+}
+
+/// Start 2 zebrad nodes using the same state directory, but different Zcash
+/// listener ports. The first node should get exclusive access to the database.
+/// The second node will panic with the Zcash state conflict hint added in #1535.
+#[test]
+fn zcash_state_conflict() -> Result<()> {
+    zebra_test::init();
+
+    // A persistent config has a fixed temp state directory, but asks the OS to
+    // automatically choose an unused port
+    let mut config = persistent_test_config()?;
+    let dir_conflict = TempDir::new("zebrad_tests")?.with_config(&mut config)?;
+
+    let mut dir_conflict_full = PathBuf::new();
+    dir_conflict_full.push(dir_conflict.path());
+    dir_conflict_full.push("state");
+    dir_conflict_full.push("state");
+    dir_conflict_full.push("v4");
+    dir_conflict_full.push("mainnet");
+
+    check_config_conflict(
+        dir_conflict.path(),
+        format!(
+            r#"Opened Zebra state cache at "{}""#,
+            dir_conflict_full.to_str().unwrap()
+        )
```

Comment:
On Windows, we get a test failure, because `\U...` is a unicode escape sequence in a regex:
https://docs.rs/regex/1.4.3/regex/#escape-sequences

```
---- zcash_state_conflict stdout ----
Error: 
   0: regex parse error:
          Opened Zebra state cache at "C:\Users\RUNNER~1\AppData\Local\Temp\zebrad_tests.kZWNTDwPyijV\state\state\v4\mainnet"
                                           ^
      error: invalid hexadecimal digit

Location:
   D:\a\zebra\zebra\zebra-test\src\command.rs:362
```

Reply:
We should make all the acceptance tests choose between an exact match and a regex pattern (#1649).

But in this PR, let's just escape this string that we want to match exactly.

We can also debug format the path, to avoid using a raw string literal - that's how we format the path in the log message:
https://github.com/ZcashFoundation/zebra/pull/1535/files#diff-a28821a2d5940d42330025f114d14455f2286d19fb3a50660c7c4b8c1f807864R48

```suggestion
        format!(
            r"Opened Zebra state cache at {:?}",
            regex::escape(dir_conflict_full.to_str().unwrap())
        )
```

Reply:
This is a bit bigger than expected so a new issue was created https://github.com/ZcashFoundation/zebra/issues/1654

In this PR we will be only checking the full path for unix. https://github.com/ZcashFoundation/zebra/pull/1535/commits/8f3c0f4663c74f2e8b562db1f438bf05a85c20a6